### PR TITLE
Adding TLS termination to gateway docs

### PR DESCRIPTION
**Pull request by willdavsmith**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### CODE REVIEW

File: docs/content/reference/resource-schema/core-schema/gateway/index.md [hunk 1]
```diff
@@ -49,9 +49,12 @@ You can define hostname information for how to access your application. See [bel
 
 | Key  | Required | Description | Example |
 |------|:--------:|-------------|---------|
-| sslPassthrough | n | Configures the gateway to passthrough encrypted SSL traffic to an HTTP route and container. Requires a single route to be set with no 'path' defined (just destination). Not compatible with path-based routing. Defaults to 'false'. | `true`
+| sslPassthrough | n | Configures the gateway to passthrough encrypted SSL traffic to an HTTP route and container. Requires a single route to be set with no 'path' defined (just destination). With sslPassthrough set to `true`, the gateway can only support SNI routing. Path based routing cannot be supported. Defaults to 'false'. | `true`
+| hostname | n | The hostname for TLS termination. | `'hostname.radapp.dev'`
+| certificateFrom | n | The Radius SecretStore resource ID that holds the TLS certificate data for TLS termination. | `secretstore.id`
```
Comment (willtsai) [inline docs/content/reference/resource-schema/core-schema/gateway/index.md]:
```suggestion
| certificateFrom | n | The Radius Secret Store resource ID that holds the TLS certificate data for TLS termination. | `secretstore.id`
```

File: docs/content/reference/resource-schema/core-schema/gateway/index.md
Comment (willtsai) [inline docs/content/reference/resource-schema/core-schema/gateway/index.md:1]: Need to add a reference to the TLS object in the Properties table at [line 27](https://github.com/project-radius/docs/pull/574/files#diff-b27ba91decd744eaabea34f506883fbb033468c17bd3f07de2109d6966197e51R27)

**Review by willtsai — DISMISSED** [review]

File: docs/content/reference/resource-schema/core-schema/gateway/snippets/gateway.bicep [hunk 2]
```diff
@@ -32,10 +32,31 @@ resource gateway 'Applications.Core/gateways@2022-03-15-privatepreview' = {
...
 }
 //GATEWAY
 
+resource secretstore 'Applications.Core/secretStores@2022-03-15-privatepreview' = {
+  name: 'secretstore'
+  properties: {
+    application: app.id
+    data: {
+    }
+  }
```
Comment (youngbupark) [inline docs/content/reference/resource-schema/core-schema/gateway/snippets/gateway.bicep:57]: @willdavsmith  @willtsai  I assume that gateway checks the type of secret and properties. Didn't we validate it ? 

```suggestion
  type: 'certificate'
  properties: {
    application: app.id
    data: {
        'tls.crt': {}
        'tls.key': {}
    }
    resource: '<secret key name>'
  }
```

**Comment by youngbupark** [comment]
@willdavsmith As we discussed before, did we create the samples for tls termination explaining bring your own cert and let's encrypt scenarios ?

File: docs/content/reference/resource-schema/core-schema/gateway/index.md [hunk 3]
```diff
@@ -49,9 +50,12 @@ You can define hostname information for how to access your application. See [bel
 
 | Key  | Required | Description | Example |
 |------|:--------:|-------------|---------|
-| sslPassthrough | n | Configures the gateway to passthrough encrypted SSL traffic to an HTTP route and container. Requires a single route to be set with no 'path' defined (just destination). Not compatible with path-based routing. Defaults to 'false'. | `true`
+| sslPassthrough | n | Configures the gateway to passthrough encrypted SSL traffic to an HTTP route and container. Requires a single route to be set with no 'path' defined (just destination). With sslPassthrough set to `true`, the gateway can only support SNI routing. Path based routing cannot be supported. Defaults to 'false'. | `true`
+| hostname | n | The hostname for TLS termination. | `'hostname.radapp.dev'`
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/core-schema/gateway/index.md:54]: Would this ever be different than `fullyQualifiedHostname` from properties.hostname?

Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/gateway/index.md:54]: @youngbupark @willdavsmith

Comment (youngbupark) [reply docs/content/reference/resource-schema/core-schema/gateway/index.md:54]: `hostname` should be a fully qualified name. Do you want to rename it ?

Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/gateway/index.md:54]: I'm more wondering if the field is redundant. A user would need to type this if I'm reading it correctly:

```bicep
resource gateway 'Applications.Core/gateways' = {
  name: 'mygateway'
  properties: {
    application: app.id
    hostname: {
      fullyQualifiedHostname: 'my.domain.com'
    }
    tls: {
      hostname: 'my.domain.com' // Is this redundant to the above hostname?
      certificateFrom: secretstore.id
    }
  }
}
```

Why couldn't they just type:

```bicep
resource gateway 'Applications.Core/gateways' = {
  name: 'mygateway'
  properties: {
    application: app.id
    hostname: {
      fullyQualifiedHostname: 'my.domain.com'
    }
    tls: {
      certificateFrom: secretstore.id
    }
  }
}
```

Would there ever be a case where they're different? 

```bicep
resource gateway 'Applications.Core/gateways' = {
  name: 'mygateway'
  properties: {
    application: app.id
    hostname: {
      fullyQualifiedHostname: 'my.domain.com'
    }
    tls: {
      hostname: 'your.domain.com' // Would this ever be different than above?
      certificateFrom: secretstore.id
    }
  }
}
```

Comment (youngbupark) [reply docs/content/reference/resource-schema/core-schema/gateway/index.md:54]: Ah. I wasn't aware of this duplication. @willdavsmith Do we need `tls.hostname` here? I think `hostname.fullyQualifiedHostname` is enough.

Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/gateway/index.md:54]: Will approve for now and make sure there's an Issue open to track consolidation. Thanks!

Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/gateway/index.md:54]: https://github.com/project-radius/radius/issues/5603

File: docs/content/reference/resource-schema/core-schema/gateway/snippets/gateway.bicep (hunk 2 above)
Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/gateway/snippets/gateway.bicep:57]: Should this type be above properties or within it?

Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/gateway/snippets/gateway.bicep:57]: Closing as this isn't displayed to the user, and is only included so `secretStore.id` is available

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**